docs/developer.md

```diff
@@ -0,0 +1,188 @@
+them to finish.
+
+If you are making routine changes for the weekly release and believe no reviews
+are necessary, you can skip the reviews (e.g., the changes are simply konflux
+reference udpates).
+
+In case you are making changes only to the configuration file, you can skip to
+the [stage run step](./developer.md#stage-run-to-check-the-new-build)
+
+
+### Wait until Konflux automatic build finishes
+
+The status of the Konflux build job can be checked in github via the 'Actions'
+tab, or in the PR checks after the merge event. The build job is called
+`mintmaker-on-pull-request`.
+
+When it is finished, Konflux will push the new image to Quay, so you can watch
+it to check when the job is done as well. Images are labelled with the git commit
+sha, so you can use that to identify which image corresponds to which commit.
+
+
+### Stage run to check the new build
+
+After the new image is built and pushed, wait until the next run and allow the
+staging environment to pick up the new image. The **Staging** environment uses
+the `latest` tag of [mintmaker-renovate-image](https://github.com/konflux-ci/mintmaker-renovate-image/),
+referenced in [manager_patches.yaml](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/mintmaker/staging/base/manager_patches.yaml).
+The PRs for this environment are created automatically.
+
+Since the `latest` tag is synamic, be mindful to check the image that is being
+used in the run, to avoid headaches.
+
+After the next MintMaker run starts, allow it to finish and check the logs. You
+can also watch them during the run.
+
+Check also that the pods used the correct image: look up the `image` field in
+the pod yamls, and check that it has the digest sha corresponding to your commit.
+
+In case you are making an update to the rpm-lockfile prototype, you can check
+the PRs in [this test repository](https://github.com/staticf0x/mintmaker-test/).
+This repository has only `.tekton` files and a RPM file, and you can check if
+there aren't any regressions with the prototype.
+
+If you do not wish to wait until the next automatic run, you can use the trick
+explained before to make an on-demand run. Be sure to use the image built by Konflux.
+
+
+### Make a change request to infra-deployments to push the new build to production
+
+
+MintMaker is released to Konflux through the [infra-deployments](https://github.com/redhat-appstudio/infra-deployments)
+repository. The **Production** environment uses a fixed tag and commit for the
+configuration. This is safer, and we can be sure of which image is being used at
+any given time.
+
+In order to release MintMaker to *production*, create a PR in [infra-deployments](https://github.com/redhat-appstudio/infra-deployments)
+with the following changes:
+
+In [kustomization.yaml](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/mintmaker/production/base/kustomization.yaml):
```

Comment:
I would maybe add here that in general, we should change these values to the same ones as in stage environment